peer/test_utils.go

```diff
@@ -643,3 +529,200 @@ func (m *mockMessageConn) LocalAddr() net.Addr {
 func (m *mockMessageConn) Close() error {
 	return nil
 }
+
+type testPeerParams struct {
```

Comment:
This is a friendly reminder that "params" is not a good struct name. Ask yourself why you are grouping it into a struct. How does this grouping of values function?

Reply:
How about `testPeerEnv`?

Reply:
Could you offer an explanation as to why this particular set of struct properties was selected? I suspect the answer to this naming question lies in that explanation.

Reply:
They are values used in the function that callers of the startPeer function would like to have access to for tests

Reply:
All tests? What kinds of tests? What are you testing? I encourage you to think a bit deeper than the obvious representational description.

Reply:
If you want to suggest a more descriptive name, you can go ahead, if not `testPeerEnv` seems good to me.

Reply:
call it `peerTestContext`

Reply:
I think `testPeerEnv` and `peerTestContext` are the same?

Reply:
We already have examples in other parts of the codebase like 

https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link_isolated_test.go#L13-L20

so my suggestion is to keep things consistent.